src/test/java/teammates/logic/core/NotificationsLogicTest.java

```diff
@@ -55,14 +63,43 @@ private void testGetNotification() throws Exception {
         assertNull(notifLogic.getNotification("nonexistant-notification"));
     }
 
+    private void testGetAllNotifications() throws Exception {
+        ______TS("success: retrieve all when there exist notifications");
+
+        n = getTypicalNotification();
+        notifDb.createEntity(n);
+        NotificationAttributes n1 = getTypicalNotification();
+        n1.setNotificationId("new-notification-id");
+        notifDb.createEntity(n1);
+
+        List<NotificationAttributes> retreivedNotifs = notifLogic.getAllNotifications();
+        retreivedNotifs.forEach(this::verifyPresentInDatabase);
+
+        removeNotificationsFromDb(n, n1);
+    }
+
+    private void testGetActiveNotificationsByTargetUser() throws Exception {
+        ______TS("success : valid target user");
```

Comment:
Can we also verify in this test case that they do not contain another user type? i.e Instructor

Reply:
Sure, I think this is a good suggestion!

Reply:
I will combine this suggestion with other similar feedback. So instead of testing based on newly added notifications, I will check the result against an expected list of notifications from typical data bundle.